src/backend/binaryen/wasm_expr_gen.ts

```diff
@@ -1923,108 +1942,176 @@ export class WASMExpressionGen {
-                member,
+            const thisTypeRef = this.wasmTypeGen.getWASMType(ownerType);
+            const valueIdxInTypeMeta = this.getTruthIdx(
+                typeMeta,
+                typeMember,
+                typeMember.hasSetter,
+            );
+            const metaRef = FunctionalFuncs.getWASMObjectMeta(
+                this.module,
                 thisRef,
-                thisTypeRef,
-                valueIdx,
-                targetValue,
             );
+            const memberNameRef = this.getStringOffset(typeMember.name);
+            let flag = ItableFlag.UNKNOWN;
+            if (typeMember.hasSetter) {
+                flag = ItableFlag.SETTER;
+            }
+            const indexRef = this.getPropIndexFromObj(
+                metaRef,
+                memberNameRef,
+                flag,
+            );
+            const flagRef = this.getPropFlagFromObj(
+                metaRef,
+                memberNameRef,
+                flag,
+            );
+            const propTypeIdRef = this.getPropTypeFromObj(
+                metaRef,
+                memberNameRef,
+                ItableFlag.UNKNOWN,
+            );
+            if (typeMeta.isInterface) {
+                return this.setInfcMember(
+                    typeMember,
+                    ownerType,
+                    thisRef,
+                    valueIdxInTypeMeta,
+                    targetValue,
+                    metaRef,
+                    indexRef,
+                    flagRef,
+                    propTypeIdRef,
+                );
+            } else {
+                return this.setObjMember(
+                    typeMember,
+                    thisRef,
+                    thisTypeRef,
+                    valueIdxInTypeMeta,
+                    targetValue,
+                );
+            }
         }
     }
 
-    private setInfcFieldWithSetter(
+    private setInfcMember(
         member: MemberDescription,
         infcType: ValueType,
         thisRef: binaryen.ExpressionRef,
-        fieldIdx: number,
+        valueIdx: number,
         targetValue: SemanticsValue,
+        metaRef: binaryen.ExpressionRef,
+        indexRef: binaryen.ExpressionRef,
+        flagRef: binaryen.ExpressionRef,
+        propTypeIdRef: binaryen.ExpressionRef,
     ) {
-        let res: binaryen.ExpressionRef;
-        const isSet = member.hasSetter ? false : true;
-        const callMethod = member.hasSetter ? true : false;
-        const targetValueRef = member.hasSetter
-            ? undefined
-            : this.wasmExprGen(targetValue);
-        const memberValueType = member.hasSetter
+        const propType = member.hasSetter
             ? (member.setter as VarValue).type
             : member.valueType;
-        res = this.getOriObjInfoByFindIdx(
+
+        const targetValueRef = this.wasmExprGen(targetValue);
+
+        /* TODO: workaround: quick path may fail, since cast failure */
```

Comment:
When will this happen?

Reply:
https://github.com/wasm-micro-runtime/ts2wasm/issues/1000

`method` in interface store in vtable, and in class store in instance.